### PR TITLE
Update packaging

### PR DESCRIPTION
**Pull request by GDYendell**
- Move config to setup.cfg
- Fix cmake find error handling

### CODE REVIEW

File: control/setup.py [hunk 1]
```diff
@@ -1,248 +1,198 @@
...
-    fem_api_include_path=os.path.join(fem_api_path, 'include')
-    
-    fem_api_sources = [fem_api_wrapper_source] 
-    
-    fem_api = Extension('excalibur.fem_api',
+
+    fem_api_path = os.path.join(fem_api_extension_path, "api")
+    fem_api_source_path = os.path.join(fem_api_path, "src")
+    fem_api_include_path = os.path.join(fem_api_path, "include")
+
```
Comment (ajgdls) [inline control/setup.py:47]: Lots of changes here which are just style alterations.  Are these all due to running a tool against the code (Black)?

**Review by ajgdls — COMMENTED** [review]
I notice that the static html has been moved, has this been reflected in the build to ensure the web interface is accessible when built?  I couldn't immediately see the change, but perhaps we have that file only in a DLS specific branch.

File: control/setup.py (hunk 1 above)
Comment (GDYendell) [reply control/setup.py:47]: Argh, I did originally do a separate commit of just running black, but I must have squashed it. Yes the majority of this diff is just black formatting. I will try to unpick that again.

File: control/setup.cfg [hunk 2]
```diff
@@ -1,17 +1,58 @@
...
+packages = find:
+package_dir =
+    =src
+
+python_requires = >=3.7
+
+install_requires =
+    odin-control
+    configparser
+    enum34
```
[the remark this answers is not in the document]
Comment (GDYendell) [reply control/setup.cfg:27]: I did not realise this is a back compatilibity package. I suppose so, but I guess some code will have to be updated too.

**Comment by timcnicholls** [comment]
I don't understand the advantage of the `excalibur_control` entry point, since you still need to pass `odin_control` a valid configuration file that specifies the adapters to load. A custom entry point wouldn't add any extra capability? 

There are use cases for the control-side (e.g. standalone test applications for f/w development/testing, possibly calibration routines used by the detector group) that use the API but not a full odin-control/odin-data infrastructure. So any additional dependencies added to the core package complicate matters. Composing a system of multiple adapters across both control and data seems like a deployment configuration step to me.

**Comment by GDYendell** [comment]
The advantage is that we can install the wheel in our DLS infrastructure creating a `prefix/bin/excalibur_control` entry point that runs within a virtual environment with odin-control, odin-data and excalibur-detector installed, so that we can just run `excalibur_control --config ...` and it can find all the adapters that it needs.

I agree. I am happy to leave this as is (requiring only odin-control) in master and make this a DLS deplyoment detail.

**Comment by GDYendell** [comment]
I realised I didn't rename the actual module directory name `excalibur` -> `excalibur_detector`. I have pushed that changed and tidied a few bits.

**Comment by GDYendell** [comment]
> I notice that the static html has been moved, has this been reflected in the build to ensure the web interface is accessible when built?

At DLS we can deploy the static directory with the tools build and then generate full paths to it in the generated config, as we do currently. The DLS build has been updated to account for the new location.

**Review by timcnicholls — APPROVED** [review]
Haven't done a file-by-file check but the structure is as agreed so I'm happy if you are 😉

**Pull request merged**